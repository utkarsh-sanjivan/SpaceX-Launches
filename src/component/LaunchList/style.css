.launch-list-conatainer {
    padding: 10px;
    margin: 0px 10px;
    display: grid;
    grid-template-columns: auto auto auto auto;
    grid-gap: 15px 15px;
}
