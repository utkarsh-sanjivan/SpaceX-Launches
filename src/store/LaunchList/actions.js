import { 
    GET_LAUNCH_LIST_REQUESTED,
} from './constants';

export function getLaunchList(payload) {
    return {
        type: GET_LAUNCH_LIST_REQUESTED,
        payload,
    }
}
