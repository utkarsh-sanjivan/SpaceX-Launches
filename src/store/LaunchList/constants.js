export const GET_LAUNCH_LIST_REQUESTED = 'GET_LAUNCH_LIST_REQUESTED';
export const GET_LAUNCH_LIST_SUCCESS = 'GET_LAUNCH_LIST_SUCCESS';
export const GET_LAUNCH_LIST_ERRORED = 'GET_LAUNCH_LIST_ERRORED';