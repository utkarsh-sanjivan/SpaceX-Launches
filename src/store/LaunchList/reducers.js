import { 
    GET_LAUNCH_LIST_REQUESTED,
    GET_LAUNCH_LIST_SUCCESS,
    GET_LAUNCH_LIST_ERRORED,
} from './constants';

const INITIAL_STATE = {
    launchListLoading: false,
    launchListErrored: false,
    launchList: [],
};

export function reducer ( state = INITIAL_STATE, action) {
    switch ( action.type ) {
        case GET_LAUNCH_LIST_REQUESTED:
            return {
                ...state,
                launchListLoading: true,
                launchListErrored: false,
                launchList: [],
            };

        case GET_LAUNCH_LIST_SUCCESS:
            return {
                ...state,
                launchListLoading: false,
                launchListErrored: false,
                launchList: action.payload,
            };

        case GET_LAUNCH_LIST_ERRORED:
            return {
                ...state,
                launchListLoading: false,
                launchListErrored: true,
                launchList: [],
            };

        default: {
            return {
                ...state
            }
        }

    };
};