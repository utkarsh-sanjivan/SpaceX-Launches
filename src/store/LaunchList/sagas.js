import { all, call, put, takeLatest } from 'redux-saga/effects';
import { 
    GET_LAUNCH_LIST_REQUESTED,
    GET_LAUNCH_LIST_SUCCESS,
    GET_LAUNCH_LIST_ERRORED,
} from './constants';
import {
    getLaunchListAPI,
} from '../../api/api.launch';

function* getLaunchListSaga(action) {
    try {
        const responseBody = yield call(getLaunchListAPI, action.payload);
        yield put({
            type: GET_LAUNCH_LIST_SUCCESS,
            payload: responseBody,
        });
    } catch (err) {
        const payload = {
            status: err.request.status,
            message: err.message,
        }
        yield put({
            type: GET_LAUNCH_LIST_ERRORED,
            payload,
        });
    };
};

export default function* userSaga() {
    yield all([
        takeLatest(GET_LAUNCH_LIST_REQUESTED, getLaunchListSaga),
    ]);
};