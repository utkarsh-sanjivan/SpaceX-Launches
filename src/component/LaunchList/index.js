import React from 'react';
import SpaceXCard from './../SpaceXCard/index.js' ;
import './style.css';

const LaunchList = props => (
    <div className="launch-list-conatainer">
        {props.launches.map(launchDetails => <SpaceXCard launchDetails={launchDetails}/>)}
    </div>
)

export default LaunchList;