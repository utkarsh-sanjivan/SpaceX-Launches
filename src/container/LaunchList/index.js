import React from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import * as LaunchListActions from '../../store/LaunchList/actions';
import './style.css';

class LaunchList extends React.Component {
  constructor (props) {
    super(props);
  }

  render () {
    return (
      <div>Launch List</div>
    );
  }
}

const mapStateToProps = ( store ) => ({
  launchList: store.launchList,
});

function mapDispatchToProps(dispatch) {
    return {
      launchListActions: bindActionCreators(LaunchListActions, dispatch),
    }
}

export default  connect(mapStateToProps, mapDispatchToProps)(LaunchList);
