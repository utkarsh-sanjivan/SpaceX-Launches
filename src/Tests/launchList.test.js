import React from 'react';
import { create } from "react-test-renderer";
import LaunchList from './../component/LaunchList/index';

test('Test SpaceX Launch List Component', () => {
  const component = create(<LaunchList 
        launches={[]}
        filterClicked={() => {}}
    />);
  expect(component.toJSON()).toMatchSnapshot()
});