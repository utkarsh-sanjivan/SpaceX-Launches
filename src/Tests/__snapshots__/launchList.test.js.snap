// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SpaceX Launch List Component 1`] = `
<div
  className="launch-list-conatainer"
/>
`;
